ingest-app/docs/api.md

```diff
@@ -107,10 +115,18 @@ The user id header allows specifying the user-id to use when saving or deleting
 
 #### <a name="CMR-Request-Id-header"></a> cmr-request-id
 
-This header returns the unique id assigned to the request. This can be used to help debug client errors. The value is a long string of the form
+This header returns the value passed in through CMR-Request-Id request header or X-Request-Id request header or a unique id generated for the client request when no value is passed in, This can be used to help debug client errors. The generated value is a long string of the form
 
     828ef0b8-a876-4579-85db-3cc9d1b5f6e5
 
+#### <a name="X-Request-Id-header"></a> x-request-id
+
+This header returns the value passed in through CMR-Request-Id or X-Request-Id request header or a unique id generated for the client request when no value is passed in, This can be used to help debug client errors. The generated value is a long string of the form
+
+    828ef0b8-a876-4579-85db-3cc9d1b5f6e5
+
+Note: X-Request-Id response header always contains the same value as the CMR-Request-Id response header.
```

Comment:
If I pass in X-Request-Id and Cmr-Request-Id that are different, do I get a 400 Bad Request?

Reply:
No. CMR-Request-Id takes precedence.  There is no need for the client to pass in both headers, but there is no harm if they want to. So I don't want to add extra code to fail this case, as long as we clearly document it.